cpp/src/gandiva/filter.cc

```diff
@@ -118,11 +135,15 @@ Status Filter::Make(SchemaPtr schema, ConditionPtr condition,
   // Return if the expression is invalid since we will not be able to process further.
   ExprValidator expr_validator(llvm_gen->types(), schema);
   ARROW_RETURN_NOT_OK(expr_validator.Validate(condition));
-  ARROW_RETURN_NOT_OK(llvm_gen->Build({condition}, SelectionVector::Mode::MODE_NONE));
+  ARROW_RETURN_NOT_OK(llvm_gen->Build({condition}, SelectionVector::Mode::MODE_NONE, obj_cache)); // to use when caching only the obj code
 
   // Instantiate the filter with the completely built llvm generator
   *filter = std::make_shared<Filter>(std::move(llvm_gen), schema, configuration);
-  cache.PutModule(cache_key, *filter);
+
+  filter->get()->SetCompiledFromCache(llvm_flag); // to use when caching only the obj code
```

Comment:
I think that you are using many `// to use when caching only the obj code` comments. Take a look and try to keep only the ones that are important to understand the code purpose

Reply:
I cleaned up this comments.